### PR TITLE
Permissions files restricted view

### PR DESCRIPTION
**Pull request by avgils**

### CODE REVIEW

File: delft3dgtprotectedservices/test_views.py [hunk 1]
```diff
@@ -30,22 +34,60 @@ def setUp(self):
         )
         assign_perm("view_scene", self.user_foo, self.scene)
 
+        self.user_foo.user_permissions.add(
```
Comment (evetion) [inline delft3dgtprotectedservices/test_views.py]: I believe this is not required. We're using an object based permissions model, so we only need to add it once as you do on r40.

File: delft3dgtprotectedservices/test_views.py [hunk 2]
```diff
@@ -30,22 +34,60 @@ def setUp(self):
...
+        # create Scene instance and assign permission for user bar
+        self.scene2 = Scene.objects.create(
+            suid="8bfdf64a-dc3f-4f48-9bf6-507c2b4e4bd9",
+            name="Test workflow 2",
+            owner=self.user_bar,
+            shared="w",
+            phase=Scene.phases.fin
+        )
+
+        self.user_bar.user_permissions.add(
```
Comment (evetion) [inline delft3dgtprotectedservices/test_views.py]: Same as above.

File: delft3dgtprotectedservices/views.py [hunk 3]
```diff
@@ -24,8 +25,13 @@ def files(request, simulation_uuid, loc):
...
     scene = get_object_or_404(Scene, suid=uuid)
 
+    loc_path, loc_extension = os.path.splitext(loc)
+
     # return 403 if not allowed
-    if not request.user.has_perm("view_scene", scene):
+    if not request.user.has_perm("restricted_view_scene", scene):
+        return HttpResponse(status=403)
+
+    if not request.user.has_perm("view_scene", scene) and loc_extension != '.png':
```
Comment (evetion) [inline delft3dgtprotectedservices/views.py]: These are switched now right? If you have the `view_scene` permission, you can view everything. The `restricted_view_scene` is only allowed to view pictures.

If you switch them, the negatives won't work. Maybe better, also to clear up to what the `not` applies to, to write it in a positive manner (if user as permission x or permission y do x else forbid).

**Review by evetion — CHANGES_REQUESTED** [review]

File: delft3dgtprotectedservices/views.py (hunk 3 above)
Comment (avgils) [reply delft3dgtprotectedservices/views.py]: I think the logic is correct:
- if you don't have restricted_view permission -> 403
- in the second if (we know we have restricted_view permissions) is you don't have view permissions and extenstions is not png -> 403
- else return 200 with response.

Tests are succeeding, so if the logic is not correct, what is then problem in the tests?

File: delft3dgtprotectedservices/test_views.py (hunk 1 above)
Comment (avgils) [reply delft3dgtprotectedservices/test_views.py]: fixed

File: delft3dgtprotectedservices/test_views.py (hunk 2 above)
Comment (avgils) [reply delft3dgtprotectedservices/test_views.py]: fixed

**Pull request merged**